src/librustdoc/core.rs

```diff
@@ -557,7 +557,7 @@ fn run_global_ctxt(
     let mut krate = tcx.sess.time("clean_crate", || clean::krate(&mut ctxt));
 
     if let Some(ref m) = krate.module {
-        if let None | Some("") = m.doc_value() {
+        if m.doc_value().map(|d| d.is_empty()).unwrap_or(true) {
```

Comment:
This will never be true because you return `None` if the value is empty. Not sure why rustdoc does that tbh, why not just check it in the place it's used?

Reply:
This is waiting on a response.

Reply:
Fixed it so back to previous behavior.

Reply:
This is not the previous behavior: you can make one change or the other, but not both. Before, `doc_value` would return `None` if there were zero lines, and this would additionally check for a single empty line. When you originally opened the PR, doc_value returned None if there were zero lines or an empty line, and the check was no longer necessary. Now that you fixed doc_value to return Some("") on an empty line, the check is necessary again.